content/blog/fs-advent-2021-fantomas-daemon/index.md

```diff
@@ -0,0 +1,100 @@
+Unfortunately, having a fix on NuGet doesn't bring it available in your IDE.
+> The stars really have to align before newer versions of Fantomas can be integrated.
+Mainly, because the version of FCS has to be in sync with what the editor has and well that can be challenging.
+
+Another downside that the IDE controls the version of Fantomas, is that you can have formatted a file yesterday and have a different result today.
+You notice this in your source control diff and that is overall just not what you want.
+
+## Daemon mode
+
+To move beyond this limitation, we tried to rethink how the editor integration should look like.
+What if you could bring your own version of Fantomas? Well, you can do this today by installing it as a local .NET tool.
+This works, but lacks the editor integration.
+
+In 4.6 we tried to solve this problem by introducing [Fantomas.Client](https://www.nuget.org/packages/Fantomas.Client/), a companion NuGet package for editors.
+The client exposes a `FantomasService` that can execute the traditional format requests as found in `Fantomas`, but it proxies those to [fantomas-tool](https://www.nuget.org/packages/fantomas-tool/) instead.
+When formatting inside an editor, `Fantomas Client` will try and find a compatible version for the current file and spin up a daemon process of `fantomas-tool`.
+This daemon process will act as an [LSP server](https://microsoft.github.io/language-server-protocol/) that send json messages back and forward to the `FantomasService`.
+As user of an editor, you don't need to worry about `Fantomas.Client`, just know that the editor will use a locally or globally installed version of `fantomas-tool`.
+Meaning, you control the version and you upgrade at your own pace.
+
+At the time of writing, there is support for VSCode (as part of the Ionide plugin).
+If you have Ionide installed, it will prompt you to install `fantomas-tool` if no compatible version can be found.
+After installing, you should be good to go as demonstrated in this video:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/1jb6K_HZaEY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Post-Daemon mode
+
+My hopes are that we can ship bug fixes to Fantomas at a high pace. If it gets on NuGet, it can be used in your IDE.
+Ever since 4.5, I have been releasing revisions on a frequent pace and people seem to have faith in this.
+Download numbers are significantly higher than say the first alpha of 4.4. Alpha and beta just don't have a stable ring to it for most people.
+I completely understand this, and that is why I'd like to move forward to a model were bugfixes are considered stable and people should not fear to upgrade.
+There is truth to this as these revisions mostly contain a fix for problem and nothing more.
```

Comment:
Maybe take this sentence out? Seems redundant to previous sentence and is a little awkwardly phrased.
If you want to keep it, another option could be: "Past revisions have mostly contained fixes for problems, and nothing that would make the update unstable."